### DISCUSSION TITLE
Incorporating Direct Routing into the system

### DISCUSSION THREAD

**Discussion by Laki4Ever79** [microsoft/call-center-ai#375] [General]
Hello @clemlesne, could this system be set up using "Direct routing" functionality of Communication Services resource? For instance, if a Country is unsupported by Microsoft's phone number availability, this would be the only option. I'm presuming config.yaml would need to be handled in a different manner?

Kind regards,
Aleksa

**Comment by clemlesne** [comment]
Hello, absolutely yes, you would require to make changes in Azure Communication Services. [You can find more doc in Microsoft Learn.](https://learn.microsoft.com/en-us/azure/communication-services/concepts/telephony/direct-routing-provisioning)

**Reply by Laki4Ever79** [reply]
Awesome. Thank you!

**Comment by Laki4Ever79** [comment]
Hello @clemlesne, I was not able to get the system working with Azure Direct routing. When I run the call-center agent, calls simply fall through and get the "unallocated number" error from the operator. Do you perhaps now if there's additional components needed to make this system work with Azure Direct routing? Is SIP server need to run in parallel for system to work, for example?

**Reply by clemlesne** [reply]
Did you followed [the doc there](https://learn.microsoft.com/en-us/azure/communication-services/concepts/telephony/direct-routing-provisioning)?

Also, the portal allows you to [see the connection status](https://learn.microsoft.com/en-us/azure/communication-services/concepts/telephony/direct-routing-provisioning#session-border-controller-connection-status) between Azure and your SBC.
